SWIPs/swip-message_to_honey.md

```diff
@@ -26,42 +25,75 @@ This SWIP is part of a series of SWIPs (but can be implemented on it's own). To
 * The design of the oracle, as well as the interaction with the oracle, should minimize the interaction of swarm nodes and the oracle maintainers with the oracle. A design proposal is clarified under the header `Technical details`, and a high-level diagram is provided below. 
 ![message_pricing.svg](./../assets/swip-message_to_honey/message_pricing.svg)
 
-## Technical details
-This section describes the interaction between the nodes and the oracle in more detail. It is by no means an indication on how this feature should be implemented, the final design and implementation will be agreed upon with the community and it could differ completely from what is described here.
-* Querying the oracle will return a `honeyPrices` object, specifying a `Time to Live (TTL)`in seconds and an array of `price` objects, each index specifying a `validFrom` field as an UTCDate and all messages with their respective prices. Taken together, the oracle returns: 
-```typescript
-{honeyPrices: 
+### Technical details
+This section describes the interaction between the nodes and the oracle in more detail.
+* Querying the oracle will return a `messagePrices` object, specifying a `Time to Live (TTL)` in seconds and `prices` object objects. The `prices` object contains an entry for each messageType (`swarmMessage`) and `swarmMessage` maps a `validFrom` to a respective `price`. Taken together, the oracle returns: 
+```json
+{
+  "messagePrices": 
   {
-    TTL, 
-    prices: [0: {validFrom, msgPrices[]}, n:{...}]
+    "TTL": <TTLValue>, 
+    "prices": {
+      "swarmMessageX": {
+        "validFrom0": <price0>, 
+        ...,                
+        "validFromN": <priceN>
+      }
+      ...,
+      "swarmMessageY": {
+        "validFrom0": <price0>,
+        ...,
+        "validFromN": <priceN>
+      }
+    }
   }
 }
 ```
-* After `TTL` expires, nodes will query the price oracle for a new `honeyPrices` object. 
-* The `TTL` is set as a state-variable in the smart-contract and can be updated.
-* To ensure that it is not possible that a new entry in the `prices` array becomes valid before all nodes are updated, the smart-contract or update-policy must ensure that the `validFrom` of new prices must be at least `TTL` seconds in the future.
-* Upon start-up, nodes will look at the contents of their local cache and query the price oracle if no `honeyPrices` object is found, or the `TTL` of their cached `honeyPrices` object has expired. 
-* If the oracle cannot be reached when the `TTL` of the last `honeyPrices` object is expired, nodes will continue to apply the `prices` as instructed by the expired `honeyObject` and will start to query the oracle at a more regular interval to re-establish connection. 
+An example of this could be:
+```json
+{
+   "messagePrices": {
+      "TTL": 3600,
+      "chunkRequest": {
+        "1566469257": 1000,             
```

Comment:
Here too, now that we can see that it's actually a timestamp, we can use it as a numerical value, not string (this will facilitate it on the parsing side, as it does not first have to convert the string into a number)

Reply:
JSON requires a string a key in an object to be a string. If you have a different proposal, I'm happy to change!